realm/src/main/java/io/realm/Realm.java

```diff
@@ -318,10 +318,8 @@ private static Realm createAndValidate(RealmConfiguration configuration, boolean
 
     private static void setupColumnIndices(Realm realm) {
         RealmProxyMediator mediator = realm.configuration.getSchemaMediator();
-        List<Class<? extends RealmObject>> modelClasses = mediator.getModelClasses();
-        int size = modelClasses.size();
-        for (int i = 0; i < size; i++) {
-            Class<? extends RealmObject> modelClass = modelClasses.get(i);
+        Set<Class<? extends RealmObject>> modelClasses = mediator.getModelClasses();
+        for (Class<? extends RealmObject> modelClass : modelClasses) {
```

Comment:
Too bad we lose the opportunity to iterate with indices, but it's a one-off operation